[Ready] Plugin manager

KrauseFx opened the pull request:
Here be 🚀 

Discussion over here https://github.com/fastlane/fastlane/issues/4744

KrauseFx committed:
Added initial Plugins.md

KrauseFx committed:
Added initial PluginManager

KrauseFx commented on fastlane/lib/fastlane/plugin_manager.rb:
```diff
@@ -0,0 +1,179 @@
+      # Using puts instead of `UI` to have the same style as the `echo`
+      puts "Installing plugin dependencies..."
+      ensure_plugins_attached!
+      with_clean_bundler_env { exec("bundle install --quiet && echo 'Sucessfully installed plugins'") }
```
@mufrtak Note how I added the `echo` to show a success message

jinjorge replied:
Successfully !Sucessfully

KrauseFx committed:
Minor improvements

KrauseFx committed:
Renamed Plugins to Pluginfile

KrauseFx committed:
Added note about storing those files in version control

KrauseFx committed:
Make use of bundler to detect what plugins are currently being loaded

jinjorge commented on fastlane/lib/fastlane/plugin_manager.rb:
```diff
@@ -0,0 +1,196 @@
+    def update_dependencies!
+      puts "Updating plugin dependencies..."
+      ensure_plugins_attached!
+      with_clean_bundler_env { exec("bundle update --quiet && echo 'Sucessfully updated plugins'") }
```
Successfully !Sucessfully

KrauseFx committed:
Fixed spelling mistake

Thanks @jinjorge

KrauseFx committed:
Added tests for PluginManager

jcampbell05 commented on fastlane/docs/Plugins.md:
```diff
@@ -0,0 +1,64 @@
+
+### Submitting the action to the fastlane main repo
+If you think your plugin will be useful to other developers as well, clone fastlane, copy your local action to fastlane/lib/fastlane/actions and submit a pull request.
+**TODO**: Add more information about what kind of PRs we want to merge
```
Should we setup an issue to discuss this ?

KrauseFx replied:
Hey @jcampbell05, great idea, I just created https://github.com/fastlane/fastlane/issues/4744 👍

jcampbell05 replied:
Okay based on my comment for this todo what about this text:

```
Please be aware we may not accept all actions submitted to be included with fastlane.

This so we can focus on those that are:

- Universal: Bundled actions should work for tools that are typically installed on a developers machine such as Xcode or Android Studio rather than unique or edge case setups.

We reccomend custom actions or plugin actions for any unique setup you may have.

- Commonly Needed: Bundled actions should fulfill a common need such as generating certificates, bumping build versions or building the app.

We reccomend using custom actions or writing the steps In the fastfile for any one-off or unique actions you wish to achieve.
```

orta replied:
You can use Danger to check for PRs to those folders and warn them too

mfurtak commented on fastlane/lib/fastlane/plugin_manager.rb:
```diff
@@ -0,0 +1,201 @@
+
+      attach_plugins_to_gemfile!(path_to_gemfile)
+      UI.success("Successfully modified '#{path_to_gemfile}'")
+      return true
```
I don't think we ever care about the return value of this method, so I would suggest we take out the booleans that are describing success.

KrauseFx replied:
Done 👍

KrauseFx committed:
Removed return value for `setup` and `ensure_plugins_attached!` methods

Fixes https://github.com/fastlane/fastlane/pull/4722#discussion_r63905341

Commit:
Initial work on plugin generator

Commit:
Add more structure and tests around plugin generation

KrauseFx committed:
Temporary version bump

KrauseFx committed:
`bundle update` is now no quiet if verbose is on

Commit:
Add tests to specifically test validation/fixing behavior

KrauseFx committed:
Introduced new `available_gems` method

Fixed issue with initial setup

KrauseFx committed:
Added more tests

Commit:
Minimal spec for PluginGeneratoer

Commit:
Create spec lets

KrauseFx committed:
Updated and extended docs

Commit:
One-time initialization in plugin generator spec

Commit:
Begin templating files as part of plugin generation

Test the correctness of the gemspec by evaluating it and
checking its values

Commit:
Fix hound warnings about exists?

Commit:
Fix plugin_generator_ui_spec

KrauseFx committed:
Updated from `fastlane_` to `fastlane-plugin` prefix

KrauseFx committed:
Ask the user for the plugin name if none is provided

KrauseFx committed:
Only ask for user input if no parameter was given

KrauseFx committed:
Fixed tests after recent change

KrauseFx committed:
Added automatic detection if plugin is available on RubyGems and offer user a way to select custom path / url

Added tests

KrauseFx committed:
Added generating of available plugins based on RubyGems API ❤️

KrauseFx committed:
Fixed code style issue

Commit:
Address feedback, adapt to new prefix, fix paths

* Improve ui object mocking to prevent test hangs if an
  interactive method is not properly mocked
* Fix file generation paths to be correctly start inside
  the gem name directory
* Fix gemspec evaluation during tests to account for its
  new location relative to the temp directory
* Fix plugin info collection to be aware of the changed
  fastlane plugin prefix value
* PluginInfo can generate the require_path from the gem_name
* Command name changed to 'new_plugin'

KrauseFx committed:
Added collecting of third party plugins

- Added tracking of third party plugins
- Added detection of reference between plugin and action
- Added tests

KrauseFx committed:
[fastlane_core] Prepare for fastlane plugins

- This adds the foundation for 3rd party plugins
- Improved code style

KrauseFx committed:
Moved fastlane_core tests to other PR

KrauseFx committed:
Added tests

mfurtak committed:
Clean up/suppress rubocop violations

KrauseFx committed:
Added documentation for plugin source

KrauseFx committed:
Fixed plugin manager not using the correct module

KrauseFx committed:
Improved plugin docs

KrauseFx committed:
Merge branch 'prepare-for-plugins' into plugin-manager

KrauseFx committed:
Skip development dependencies by default when installing plugins

KrauseFx committed:
Removed dead test that didn't actually test things

KrauseFx committed:
Added proper catching of SystemExit and similar to cleanly finish fastlane run

Tested with `xcake`, which throws a `Xcake::Informative` object

KrauseFx committed:
Also track crash when a system exit is used

KrauseFx committed:
Fixed code style

Commit:
Add email collection to the plugin generator interview

Commit:
Pre-fill summary, description, homepage in gemspec; reword info collectory questions

KrauseFx committed:
Removed bundler code from bin/fastlane

Commit:
LICENSE creation

KrauseFx committed:
Added more plugins docs

KrauseFx committed:
Added docs for `new_plugin` for existing gems

Commit:
Add support for collecting gemspec summary and description

jcampbell05 committed:
Plugin manager (#4776)

* Handle old bundled actions.

* Check on one off as well.

* Update message.

KrauseFx committed:
Added tests for previously bundled fastlane actions

Commit:
Add missing LICENSE file

Commit:
Accept plugin name during command line call

jcampbell05 commented on fastlane/spec/plugin_manager_spec.rb:
```diff
@@ -118,6 +118,27 @@
+          end").runner.execute(:test)
+        end.to raise_exception("The action '#{deprecated_action}' is no longer bundled with fastlane. You can install it using `fastlane add_plugin deprecated`")
+      end
+    end
```
Can we add a test to verify it doesn't show this message if the plugin is installed or do you think this case is covered @KrauseFx  ?

KrauseFx replied:
I'll take a look

jcampbell05 commented on fastlane/docs/Plugins.md:
```diff
@@ -30,6 +30,28 @@ What will this do
+
+# Specify a version requirements
+gem "fastlane-plugin-xcversoin", "1.1.0"
+gem "fastlane-plugin-xcversoin", ">= 1.0"
```
Should be xcversion :P

suculent replied:
Thank you! Actually helped solving my other issue. :o)

> On 25 May 2016, at 17:37, James Campbell <notifications@github.com> wrote:
> 
> In fastlane/docs/Plugins.md <https://github.com/fastlane/fastlane/pull/4722#discussion_r64595282>:
> 
> > +Your `fastlane/Pluginfile` contains the list of all `fastlane` plugins your project uses. The `Pluginfile` is a [Gemfile](http://bundler.io/gemfile.html) that gets imported from your main `Gemfile`.
> > +You specificy all dependencies, including the required version numbers:
> > +
> > +```ruby
> > +# Fetched from RubyGems.org
> > +gem "fastlane-plugin-xcversion"
> > +
> > +# Fetched from GitHub
> > +gem "fastlane-plugin-xcversion", git: "https://github.com/fastlane/fastlane-plugin-xcversion"
> > +
> > +# Fetched from a local directory
> > +gem "fastlane-plugin-xcversion", path: "https://github.com/fastlane/fastlane-plugin-xcversion"
> > +
> > +# Specify a version requirements
> > +gem "fastlane-plugin-xcversoin", "1.1.0"
> > +gem "fastlane-plugin-xcversoin", ">= 1.0"
> Should be xcversion :P
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub <https://github.com/fastlane/fastlane/pull/4722/files/9525407571e7fecf14605476e40d42bbe6dcb463..e4328092a37bb0a9a188c2075d7349dda808ef48#r64595282>

KrauseFx replied:
Thanks, that's fixed now

KrauseFx replied:
I see the example path was also wrong

Commit:
Define the plugin action class as part of the templating

Commit:
Make the plugin gem depend on fastlane

Commit:
Make fastlane a development dependency of the plugin bundle

* Add other needed development dependencies to the plugin bundle
  template

Commit:
Fix plugin_generator_spec

Commit:
Add helper file generation to the plugin generator

Commit:
Add .rspec file to the plugin generator content

KrauseFx committed:
Fixed spelling mistake in Plugins.md

KrauseFx committed:
Fixed sample path in Plugins.md

Commit:
Fix failing rubocop offense

KrauseFx committed:
Added test for deprecated actions when plugin is available

KrauseFx committed:
Improved plugin docs

KrauseFx committed:
Updated RubyGems search query

KrauseFx committed:
Added initial Troubleshooting guide

KrauseFx committed:
Improved output when adding a plugin

KrauseFx committed:
Print command when running fastlane install_plugins or fastlane update_plugins

KrauseFx committed:
No more --no-development since it updates the bundler config

KrauseFx committed:
Resolved merge conflict

Commit:
Add Gemfile to plugin generator content

Commit:
Add .gitignore to the plugin generator content

KrauseFx committed:
Moved all plugin classes into plugins subfolder

KrauseFx committed:
Improved wording and documentation

KrauseFx committed:
Renamed Pluginfile constant

KrauseFx committed:
Changed "pluginsfile_" to "pluginfile_"

KrauseFx committed:
Added plugin summary to see what plugins are loaded

KrauseFx committed:
Improved plugin docs

KrauseFx committed:
Refactored plugin references to also contain version number

Made determine version number a class method with alias to the other method
Added constant for troubleshooting guide

KrauseFx committed:
fastlane now uses plugins itself to run common ruby tasks

KrauseFx committed:
Enabled fastlane plugins for the fastlane's Gemfile

KrauseFx commented:
Tests are currently failing because we need to push out the changes of that were made in `fastlane_core` to determine a version number: https://github.com/fastlane/fastlane/pull/4722/commits/0a9e0351377313e5ea31cfb8fcfe8f7653e5495e#diff-861844ac6c8a585e117d3135bd82679cR136

I'll work on a separate PR and release for `fastlane_core` tomorrow, but don't worry, the tests should work just fine locally 👍

KrauseFx commented:
This is now ready for initial review 🚀

KrauseFx commented:
To test the plugin system locally, run `fastlane install_plugins` and then `fastlane test` (which will run the unit tests, using an external plugin for `rspec`, hosted [over here](https://github.com/KrauseFx/fastlane-plugin-ruby)

Commit:
Add main plugin .rb file to the plugin generator content

Commit:
Update README template for plugin with badge.

Commit:
Add a success message when the plugin generator is done

Commit:
Add spec helper generation

Commit:
Add action_spec creation to the plugin generator

Commit:
Add Rakefile generation

Commit:
Remove description from PluginInfo model

Commit:
Add more details the generated README.md

Commit:
Continue improving the documentation for plugins and generator

KrauseFx committed:
Updated fastlane_core dependency

KrauseFx committed:
Added support for hidden ".fastlane" folder

KrauseFx committed:
Added new `fastlane search_plugins` command (#4853)

* Added new `fastlane search_plugin` command

* Added tests

* Improved design of search result table

* Switched from `search_plugin` to `search_plugins`

KrauseFx committed:
Removed now: true from tests

KrauseFx committed:
Renamed fastlane logo

KrauseFx committed:
Merge pull request #4762 from fastlane/plugin-generator

Plugin generator

KrauseFx committed:
Switched from `capitalize` to `fastlane_class` to work with underlines too

KrauseFx committed:
Show broken plugins in plugins table

KrauseFx committed:
Prepare beta release

KrauseFx committed:
Made bundler a fastlane dependency

KrauseFx committed:
Prepare next beta release

Commit:
Optimise and hot badge.

mfurtak committed:
Merge pull request #4876 from jcampbell05/plugin-manager

Embeds and compresses fastlane logo into badge.

ohayon committed:
auto detect email and author from git config

ohayon committed:
Merge pull request #4882 from fastlane/auto-detect-email

auto detect email and author from git config

Commit:
Suppress command output for git queries during plugin info collection

mfurtak committed:
Merge pull request #4888 from fastlane/hide-git-calls

Suppress command output for git queries during plugin info collection

KrauseFx commented on fastlane/lib/fastlane/plugins/templates/README.md.erb:
```diff
@@ -0,0 +1,31 @@
+
+## Troubleshooting
+
+For some , check out the [Plugins Troubleshooting](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/PluginsTroubleshooting.md) doc in the main `fastlane` repo.
```
@mfurtak There is a spelling mistake here `For some , check `

mfurtak committed:
Correct incomplete sentence in plugin generator README

mfurtak committed:
Reimplement the PluginGenerator based on dir templating

Moves all of the plugin generator template files from a flat
directory into one that is actually shaped like the intended
destination structure.

This allows the PluginGenerator to stop encoding knowledge
about the intended structure of the plugin, and just focus
on copying and templating logic.

The template directory uses %-wrapped variables to allow
directory and file names to have variables inserted at
run time. Those variable names are used as accessor calls
into the PluginInfo object to get replacement values.

The tests have been changed to use a plugin name containing
an underscore and have been corrected to check for expected
values based on fastlane_class transform of the plugin name

mfurtak committed:
Merge pull request #4901 from fastlane/plugin-templator

Reimplement the PluginGenerator based on dir templating

mfurtak committed:
Move plugin badge to root fastlane assets dir

Update the plugin README to look in this new path

mfurtak committed:
Merge pull request #4908 from fastlane/move-plugin-badge

Move plugin badge to root fastlane assets dir

KrauseFx committed:
Show error message if fastlane directory is missing

KrauseFx committed:
Integrated Club Mate plugin into fastlane setup

mfurtak commented on fastlane/docs/AvailablePlugins.md:
```diff
@@ -0,0 +1,5 @@
+
+| Plugin Name | Description | Downloads
+--------------|-------------|----------|----------
+[apprepo](https://github.com/suculent/fastlane-plugin-apprepo) | experimental fastlane plugin based on https://github.com/suculent/apprepo SFTP uploader | 111
```
Shall we run this one more time before we commit?

KrauseFx replied:
Absolutely. I was thinking even doing this every day on some CI machine that will submit a PR. 
I'd still like the docs to be available on GitHub too, so people can see the available plugins without having to install and start using `fastlane`. 
Let's start with this, and see how it goes 👍

mfurtak commented on fastlane/docs/Plugins.md:
```diff
@@ -0,0 +1,149 @@
+- Add the plugin to `fastlane/Pluginfile`
+- Make sure your `fastlane/Pluginfile` is properly referenced from your `./Gemfile`
+- Run `bundle install` to make sure all required dependencies are installed on your local machine (this step might ask for your admin password to install Ruby gems)
+
```
We should probably mention the `Gemfile.lock` and explain that it, the `Pluginfile`, and the `Gemfile` all need to be checked in to source control with project code.

KrauseFx replied:
I just noticed we should say `fastlane install_plugins` instead, right? Also running this is optional

KrauseFx replied:
However by doing this once, people are more likely to remember, so let's keep it there

mfurtak commented on fastlane/docs/Plugins.md:
````diff
@@ -0,0 +1,149 @@
+```
+cd ~/new/folder/[plugin_name]
+
+fastlane new_plugin
````
Let's note that you can pass the plugin name as a command line arg:

```
fastlane new_plugin [plugin_name]
```

KrauseFx replied:
Done 👍

mfurtak commented on fastlane/docs/Plugins.md:
````diff
@@ -0,0 +1,149 @@
+### Create your own plugin
+
+```
+cd ~/new/folder/[plugin_name]
````
Because the folder `[plugin_name]/` will be created by the command, let's leave that off here.

mfurtak commented on fastlane/docs/Plugins.md:
```diff
@@ -0,0 +1,149 @@
+All you have to do if you have an exiting gem:
+
+- Navigate to your gem
+- `fastlane new_plugin`
```
`fastlane new_plugin [plugin_name]`

KrauseFx replied:
Done 👍

mfurtak commented on fastlane/lib/fastlane/action_collector.rb:
```diff
@@ -15,7 +26,28 @@ def show_message
+    #   :xcversion
+    #   "fastlane-plugin-my_plugin/xcversion"
+    def self.determine_version(name)
+      return super(name) if super(name)
```
Let's assign the value of `super(name)` to a variable so that we don't make the method call 2x

KrauseFx replied:
I wonder if there is a Ruby operator for this, since it's quite common. I updated the code 👍

mfurtak commented on fastlane/lib/fastlane/commands_generator.rb:
```diff
@@ -170,8 +171,64 @@ def run
+      #####################################################
+
+      command :new_plugin do |c|
+        c.syntax = 'fastlane create_plugin [plugin_name]'
```
'create_plugin' -> 'new_plugin'

KrauseFx replied:
👍

mfurtak commented on fastlane/lib/fastlane/plugins/plugin_manager.rb:
```diff
@@ -0,0 +1,338 @@
+      plugin_name = self.class.plugin_prefix + plugin_name unless plugin_name.start_with?(self.class.plugin_prefix)
+
+      unless plugin_is_added_as_dependency?(plugin_name)
+        content = pluginfile_content || "# Autogenerated by fastlane\n\n"
```
Consider adding a message like "Ensure this file is checked in to source control!"

KrauseFx replied:
Great idea

mfurtak commented on fastlane/lib/fastlane/plugins/plugin_manager.rb:
```diff
@@ -0,0 +1,338 @@
+
+    # Modify the user's Gemfile to load the plugins
+    def attach_plugins_to_gemfile!(path_to_gemfile)
+      content = gemfile_content || GEMFILE_SOURCE_LINE
```
Consider adding a message to the file like "Ensure this file is checked in to source control!" if the file is being newly generated.

mfurtak commented on fastlane/lib/fastlane/plugins/plugin_manager.rb:
```diff
@@ -0,0 +1,338 @@
+      UI.verbose("Checking if there are any plugins that should be loaded...")
+
+      loaded_plugins = false
+      Gem::Specification.each do |current_gem|
```
I believe that this will not find plugins which are specified in Bundler via `git:` 

http://bundler.io/git.html

> Because RubyGems lacks the ability to handle gems from git, any gems installed from a git repository will not show up in `gem list`.

KrauseFx replied:
Wow, great catch, you're right! I wasn't aware we use 2 different systems for that. So using 
```ruby
dsl = Bundler::Dsl.evaluate(gemfile_path, nil, true)
 return dsl.dependencies.map(&:name)
```
is the right way?

mfurtak replied:
Hm. I think it's the right way assuming that the Gemfile + Plugin file are the canonical authority for which plugins are available/enabled.

The `Gem::Specification.each` way only really cares if the Gem has _ever_ been installed locally. So, if you do `gem install fastlane-plugin-xcake` but never add it to your Gemfile or Pluginfile, the `Gem::Specification.each` way would work, but the `Bundler::Dsl.evaluate` way would not.

However, I don't think that is a use case we are trying to support, but `git:` hosting is!

mfurtak commented on fastlane/lib/fastlane/plugins/plugin_manager.rb:
```diff
@@ -0,0 +1,338 @@
+        end
+      end
+
+      if !loaded_plugins && PluginManager.new.pluginfile_content.to_s.include?(plugin_prefix)
```
Creating a new instance of this class for use during one if its class methods feels strange to me. If we need class level access to `pluginfile_content` we should adjust it to also be a class method.

KrauseFx replied:
That's fixed with https://github.com/fastlane/fastlane/pull/4927/files#diff-d799ecdf08be1a9f3433cc543c2d22ffR276 👍

mfurtak commented:
Finished my pass!

I have one additional question:

Is there anything to prevent plugins from defining actions that collide with action names used directly by fastlane?

KrauseFx commented:
@mfurtak Regarding your last question: No, a plugin might use the same name as an existing action. Very good point, we should prefer plugins over built-in actions. 

I talked with @joshdholtz, he builds the sentry action as a plugin now, by preferring plugins over built-in actions, Sentry can easily overwrite built-in actions with additional functionality. 

What do you think?

KrauseFx committed:
Addressed @mfurtak feedback

joshdholtz commented:
@KrauseFx I like that solution! One other question about the plugins though... What happens if somebody else creates a plugin with the same action name as mine? Can we / is it possible for plugins to be namespaced? Like... `Sentry::upload_symbols`?

KrauseFx commented:
Sorry, no, plugins are currently not namespaced, I have the feeling this would add a lot of complexity to the `Fastfile`. However that's something @orta did with [danger](https://github.com/danger/danger). 

Maybe that's something we could offer optionally, to call an action using
```ruby
upload_dsym(...)
sentry.upload_dsym(...)
```

I'm not 100% convinced that we should add complexity to the end-user just to avoid naming conflicts of actions. 
We should however show appropriate warnings if that happens, with details on what 2 plugins conflict for what action

joshdholtz commented:
@KrauseFx I completely agree with that we don't want to add complexity 👍  Just wanted to make sure there wouldn't be any confusion with conflicting names but warning sounds good to me. 

So is the best practice here for me to name my action the same? (Since fastlane will choose the plugin action over the built in action)

mfurtak commented on fastlane/docs/Plugins.md:
```diff
@@ -56,7 +56,8 @@ This will:
 - Add the plugin to `fastlane/Pluginfile`
 - Make sure your `fastlane/Pluginfile` is properly referenced from your `./Gemfile`
-- Run `bundle install` to make sure all required dependencies are installed on your local machine (this step might ask for your admin password to install Ruby gems)
+- Run `fastlane install_dependencies` to make sure all required dependencies are installed on your local machine (this step might ask for your admin password to install Ruby gems)
```
The method name is `install_dependencies`, but the fastlane command name is `install_plugins`, right?

KrauseFx replied:
Ohhh, great point, it's `install_plugins`

mfurtak commented on fastlane/lib/fastlane/plugins/plugin_manager.rb:
```diff
@@ -0,0 +1,338 @@
+
+    PLUGINFILE_NAME = "Pluginfile".freeze
+    DEFAULT_GEMFILE_PATH = "Gemfile".freeze
+    GEMFILE_SOURCE_LINE = "source \"https://rubygems.org\"\n"
```
Can we modify this string to also include the "Ensure this file is checked in to source control" line so that brand new `Gemfile`s

jcampbell05 commented:
I would take the objc approach and just prefix the action with your namespace

Commit:
Fixup "Ensure..." line in Gemfile and Pluginfile

KrauseFx committed:
Moved PluginManager object to Fastlane.plugin_manager (#4927)

* Moved PluginManager object to Fastlane.plugin_manager

* Adressed feedback

KrauseFx committed:
Renamed Plugins.md.erb to AvailablePlugins.md.erb

KrauseFx committed:
Adressed PR feedback

KrauseFx committed:
Merge branch 'plugin-manager' of https://github.com/fastlane/fastlane into plugin-manager

KrauseFx committed:
Added minimum requirement for plugins to Plugins.md

KrauseFx committed:
Added install instructions to AvailablePlugins.md

KrauseFx committed:
Fixed plugin generator calling wrong class

KrauseFx committed:
Commented out the spec.homepage

KrauseFx committed:
Updated AvailablePlugins.md

KrauseFx committed:
Moved all plugin specs into subfolder

KrauseFx committed:
Replaced `install_dependencies` with `install_plugins`

KrauseFx committed:
Improved wording

KrauseFx committed:
Minor improvements

KrauseFx committed:
Updated lane name blacklist to not allow plugin commands

orta commented on fastlane/docs/Plugins.md:
````diff
@@ -0,0 +1,152 @@
+
+```
+fastlane add_plugin [name]
+```
````
I guess you don't support namespaces, but `plugin search`, `plugin add` etc would be easier to remember ( and auto-complete )

KrauseFx replied:
Yeah, all fastlane CLIs work like this, so we'll stick with it for now 👍

orta commented on fastlane/fastlane/Fastfile:
```diff
@@ -106,6 +110,8 @@ lane :release do |options|
     )
   end
 
+  clubmate
```
heh

KrauseFx replied:
Kind of a requirement for fastlane deployments :clubmate:

KrauseFx committed:
Added auto-import of fastlane actions in plugin specs

mfurtak commented:
Let's :ship: it!

mpirri commented:
👍

KrauseFx commented:
Alright, fastlane plugins are merged and shipped with the latest `fastlane` release 🚀 Exciting times
